nixos/modules/virtualisation/docker.nix

```diff
@@ -165,7 +176,8 @@ in
           ];
         };
 
-        path = [ pkgs.kmod ] ++ (optional (cfg.storageDriver == "zfs") pkgs.zfs);
+        path = [ pkgs.kmod ] ++ (optional (cfg.storageDriver == "zfs") pkgs.zfs)
+          ++ (optional cfg.enableNvidia pkgs.nvidia-docker);
```

Comment:
Not as many parens needed:
`.. ++ optional (cfg.storageDriver == "zfs") pkgs.zfs ++ optional cfg.enableNvidia ...`